polygon/bridge/bridge.go

```diff
@@ -33,47 +33,59 @@ import (
 
 	libcommon "github.com/erigontech/erigon-lib/common"
 	"github.com/erigontech/erigon/core/types"
-	"github.com/erigontech/erigon/polygon/bor/borcfg"
 	"github.com/erigontech/erigon/polygon/heimdall"
 )
 
 type eventFetcher interface {
 	FetchStateSyncEvents(ctx context.Context, fromId uint64, to time.Time, limit int) ([]*heimdall.EventRecordWithTime, error)
 }
 
+type DBConfig struct {
+	DataDir   string
+	RoTxLimit int64
+}
+
 type Config struct {
-	DataDir      string
-	Logger       log.Logger
-	BorConfig    *borcfg.BorConfig
-	EventFetcher eventFetcher
-	RoTxLimit    int64
+	Logger                log.Logger
+	StateReceiverContract string
+	EventFetcher          eventFetcher
+
+	IsSprintStartFn           func(uint64) bool
+	CalculateSprintLengthFn   func(uint64) uint64
+	IsIndoreFn                func(uint64) bool
+	CalculateStateSyncDelayFn func(uint64) uint64
+	OverrideStateSyncRecords  map[string]int
 }
 
-func Assemble(config Config) *Bridge {
-	bridgeDB := polygoncommon.NewDatabase(config.DataDir, kv.PolygonBridgeDB, databaseTablesCfg, config.Logger, false /* accede */, config.RoTxLimit)
+func Assemble(config Config, dbConfig DBConfig) *Bridge {
```

Comment:
can move `DBConfig` inside the service `Config` using composition:

```
type Config {
    DBConfig
    ...
}
```

and

```
type ReaderConfig struct {
    DBConfig
    ...
}
```

Would be good to do the same for `heimdall.Config` and `heimdall.ReaderConfig` for consistency if you do decide to go down this path. Alternatively it is also fine to eliminate `DBConfig` and re-define `DataDir` and `RoTxLimiter` on the `Config` struct - since our DBConfig is only 2 attributes at the moment (if it was more it would have been stronger reason to separate it out).

Reply:
+1